lint-configs/.clang-tidy

```diff
@@ -25,6 +25,13 @@ Checks: >-
   -readability-simplify-boolean-expr,
 
 CheckOptions:
+  # These checks are intended to catch potentially bug-prone conversions from `signed char`  to
+  # larger integer types. However, these checks are only relevant when `signed char` is used to
+  # store character data. If a type like `int8_t` is used to represent signed integers, these checks
+  # can be safely skipped.
+  bugprone-signed-char-misuse.CharTypdefsToIgnore: "int8_t"
+  cert-str34-c.CharTypdefsToIgnore: "int8_t"
```

Comment:
Do we need the alias?

Reply:
If we don't add thi s alias into the config file, we will get the following warnings (clang-tidy 19.1):
```
/home/lzh/clp_oss/clp/components/core/src/clp/ffi/ir_stream/ir_unit_deserialization_methods.cpp:240:15: error: 'signed char' to 'value_int_t' (aka 'long') conversion; consider casting to 'unsigned char' first. [cert-str34-c,-warnings-as-errors]
  240 |         val = deserialized_val;
      |               ^
Suppressed 205776 warnings (205626 in non-user code, 150 NOLINT).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
1 warning treated as error
```
Seems like it's still required to silent `cert-str34-c`